### ISSUE TITLE
@mustCallSuper confusing with mixins

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#2600]
In the example below, to ensure that all resources of the base class `Base` are freed up when it is disposed, I am annotating its method with `@mustCallSuper`.  Now while implementing `dispose` in the sub class `Sub`, the analyzer lint `must_call_super` helpfully tells me that I need to call `super.dispose()`, which I can easily do thinking that now the resources of Base will get disposed when Sub is disposed. However, that is not the case due to the `FooMixin`: `super.dispose()` will actually call `FooMixin.dispose()` and `Base.dispose()` is never called (even though I said it must be called).

This is of course the expected behavior for how Mixins work in Dart. However I find it confusing that there's no warning that the documented `@mustCallSuper` contract is not fulfilled since `Base.dispose()` is never invoked.

From the [mustCallSuper doc](https://master-api.flutter.dev/flutter/meta/mustCallSuper-constant.html):

> Indicates that every invocation of a method that overrides m must also invoke m.

```dart
import 'package:meta/meta.dart';

void main() {
  Sub().dispose();
}

class Base {
  @mustCallSuper
  void dispose() {
    print('Base dispose');
  }
}

class Sub extends Base with FooMixin {
  @override
  void dispose() {
    print('Sub dispose');
    super.dispose();
  }
}

mixin FooMixin {
  void dispose() {
    print('mixin dispose');
  }
}
```
